src/cargo/util/lints.rs

```diff
@@ -260,63 +261,87 @@ pub fn unused_dependencies(
     if lint_level == LintLevel::Allow {
         return Ok(());
     }
-
+    let mut emitted_source = None;
     let manifest = pkg.manifest();
-    let activated_opt_deps = manifest
-        .resolved_toml()
-        .features()
-        .map(|map| {
-            map.values()
-                .flatten()
-                .filter_map(|f| match FeatureValue::new(InternedString::new(f)) {
-                    Dep { dep_name } => Some(dep_name.as_str()),
-                    _ => None,
-                })
-                .collect::<HashSet<_>>()
-        })
-        .unwrap_or_default();
+    let original_toml = manifest.original_toml();
+    let all_dependencies = manifest
+        .dependencies()
+        .into_iter()
+        .map(|d| d.name_in_toml().to_string())
+        .collect::<HashSet<String>>();
```

Comment:
We probably shouldn't be converting away from `InternedString` but converting things to it

Reply:
Though I'm unsure if this will make much of a difference, so its probably fine